gateway/gateway.go

```diff
@@ -91,10 +91,11 @@ type IPFSBackend interface {
 	// Size, and Cid.
 	Get(context.Context, ImmutablePath) (ContentPathMetadata, *GetResponse, error)
 
-	// GetRange returns a full UnixFS file object. Ranges passed in are advisory for pre-fetching data, however
-	// consumers of this function may require extra data beyond the passed ranges (e.g. the initial bit of the file
-	// might be used for content type sniffing even if only the end of the file is requested).
-	GetRange(context.Context, ImmutablePath, ...GetRange) (ContentPathMetadata, files.File, error)
+	// GetRange returns a full UnixFS file object, or a UnixFS directory. Ranges passed in are advisory for pre-fetching
+	// data, however consumers of this function may require extra data beyond the passed ranges (e.g. the initial bit of
+	// the file might be used for content type sniffing even if only the end of the file is requested).
+	// Note: there is currently no semantic meaning attached to a range request for a directory
+	GetRange(context.Context, ImmutablePath, ...GetRange) (ContentPathMetadata, *GetResponse, error)
```

Comment:
Looking at the code now, should we just have a single function `Get(context.Context, ImmutablePath, ...GetRange) (ContentPathMetadata, *GetResponse, error)` and drop `GetRange` entirely?

Reply:
I think that's a good idea.

Reply:
I will do it in https://github.com/ipfs/bifrost-gateway/pull/61